### PR TITLE
[mypy] Add some variable annotations

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/9867

Add some variable annotations to reduce mypy warnings

### CODE REVIEW

File: src/twisted/mail/test/test_smtp.py [hunk 1]
```diff
@@ -384,8 +385,8 @@ class DummyESMTP(DummyProto, smtp.ESMTP):
 
 
 class AnotherTestCase:
-    serverClass = None
-    clientClass = None
+    serverClass = None  # type: Optional[Type[DummyProto]]
```
Comment (adiroiban) [inline src/twisted/mail/test/test_smtp.py]: I think that this should be a generic protocol. and not the specific DummyProto

```suggestion
    serverClass = None  # type: Optional[Type[protocol.Protocol]]
```

File: src/twisted/mail/test/test_smtp.py [hunk 2]
```diff
@@ -384,8 +385,8 @@ class DummyESMTP(DummyProto, smtp.ESMTP):
...
+    clientClass = None  # type: Optional[Type[MyClient]]
```
Comment (adiroiban) [inline src/twisted/mail/test/test_smtp.py]: Maybe 
```suggestion
    clientClass = None  # type: Optional[Type[smtp.SMTPClient]]
```

File: src/twisted/mail/protocols.py [hunk 3]
```diff
@@ -37,7 +38,7 @@ class DomainDeliveryBase:
         Sub-classes should set this appropriately.
     """
     service = None
-    protocolName = None
+    protocolName = None  # type: Optional[bytes]
```
Comment (adiroiban) [inline src/twisted/mail/protocols.py]: I don't think that this is optional. protocolName should always be defined, as described by the docstring.

Maybe have something like this.
```suggestion
    protocolName = b'not-implemented-protocol`
```

**Review by adiroiban — APPROVED** [review]
Only minor comments. Thanks!

File: src/twisted/mail/test/test_smtp.py (hunk 2 above)
Comment (rodrigc) [reply src/twisted/mail/test/test_smtp.py]: Done

File: src/twisted/mail/protocols.py (hunk 3 above)
Comment (rodrigc) [reply src/twisted/mail/protocols.py]: Done

**Pull request merged**